src/cls/rbd/cls_rbd.cc

```diff
@@ -2675,6 +2693,51 @@ int metadata_get(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   return 0;
 }
 
+int get_snapshot_limit(cls_method_context_t hctx, bufferlist *in,
+		       bufferlist *out)
+{
+  int rc;
+  uint64_t snap_limit;
+
+  rc = read_key(hctx, "snap_limit", &snap_limit);
+  if (rc == -ENOENT) {
+    rc = 0;
+    ::encode(UINT64_MAX, *out);
+  } else {
```

Comment:
In that case, we just pass the returned value unmodified the caller. There really isn't much else we can do if read_key fails.